app/components/LineChart.tsx

```diff
@@ -57,6 +81,14 @@ export const HardwareLineChart = ({ vials, rawData, property = "raw" }) => {
               connectNulls={true}
             />
           ))}
+          {eventData.map((event) => (
+            <ReferenceLine
+              key={event}
+              x={event.timestamp}
+              label={event.data.message}
```

Comment:
Label will take a component as prop, and daisyUI provides a tooltip: https://daisyui.com/components/tooltip/,

```
label = {() => {
  return (
    <div className="tooltip" data-tip={event.data.message}>
      {event.data.message.length > 20 ? `${event.data.message.slice(0, 20)}...` : event.data.message}
    </div>
  )
}}
```

Reply:
Looks cool, but when I do this I don't see any text at all. I tried with plain div and span elements like this, doesn't seem to show. There may be some requirements it has on the underlying element...